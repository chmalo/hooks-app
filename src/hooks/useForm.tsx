import { ChangeEvent, useState } from "react"

export const useForm = (initialState: any = {}) => {
  const [values, setValues] = useState(initialState)

  const handleInputChange = ({ target }: ChangeEvent<HTMLInputElement>) => {
    console.log(target.name)
    setValues({
      ...values,
      [target.name]: target.value,
    })
  }

  return [values, handleInputChange]
}
