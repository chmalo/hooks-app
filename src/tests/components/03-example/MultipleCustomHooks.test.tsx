import { render } from "@testing-library/react";
import {describe, expect, test, vitest} from "vitest";
import {MultipleCustomHooks} from "../../../components/03-example/MultipleCustomHooks";
import {useFetch} from "../../../hooks/useFetch";
import {useCounter} from "../../../hooks/useCounter";
vitest.mock("../../../hooks/useFetch")
vitest.mock("../../../hooks/useCounter")



describe("Testing in MultipleCustomHooks", () => {

    // @ts-ignore
    useCounter.mockReturnValue({
        counter: 10,
        increment: () => {}
    })

    test("it should show correctly", () => {

        // @ts-ignore
        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null
        })

        const {container} = render(<MultipleCustomHooks />)
        expect(container).toMatchSnapshot()
    })

    // test("it should show info correctly", () => {
    //
    //     useFetch.mockReturnValue({
    //         data: [{
    //             author: 'Christian',
    //             quote: 'hola mundo'
    //         }],
    //         loading: false,
    //         error: null
    //     })
    //
    //     const {container} = render(<MultipleCustomHooks />)
    // })

})