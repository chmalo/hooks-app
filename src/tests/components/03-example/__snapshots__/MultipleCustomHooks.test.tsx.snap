// Vitest Snapshot v1

exports[`Testing in MultipleCustomHooks > it should show correctly 1`] = `
<div>
  <div>
    <h1>
      BreakibgBad Quotes
    </h1>
    <hr />
    <div
      class="alert alert-info text-center"
    >
      Loading...
    </div>
    <button
      class="btn btn-primary"
    >
      siguiente quote
    </button>
  </div>
</div>
`;
